samples/tutorial/Tutorial3.md

```diff
@@ -0,0 +1,415 @@
+# Step 3
+
+_State throughout a tree of workflows_
+
+## Setup
+
+To follow this tutorial, launch Android Studio and open this folder (`samples/tutorial`).
+
+Start from the implementation of `tutorial-2-complete` if you're skipping ahead.
+
+## Editing todo items
+
+Now that a user can "log in" to their todo list, we want to add the ability to edit the todo items listed.
+
+### State ownership
+
+In the workflow framework, data flows _down_ the tree as properties to child workflows, and actions come _up_ as output events (as in the traditional computer science sense that trees that grow downward).
+
+What this means is that state should be created as far down the tree as possible, to limit the scope of state to be as small as possible. Additionally, there should be only one "owner" of the state in the tree - if it's passed farther down the tree, it should be a copy or read-only version of it - so there is no shared mutable state in multiple workflows.
+
+When a child workflow has a copy of the state from its parent, it should change it by emitting an _output_ back to the parent, requesting that it be changed. The child will then receive an updated snapshot of the data from the parent - keeping ownership at a single level of the tree.
```

Comment:
Avoid overloading "snapshot" here. Perhaps "receive an updated copy of the data"